gpytorch/functions/__init__.py

```diff
@@ -15,16 +16,16 @@ def add_diag(input, diag):
     Adds a diagonal matrix s*I to the input matrix input.
 
     Args:
-        - input (matrix nxn) - Variable or LazyTensor wrapping matrix to add diagonal \
+        - input (matrix nxn) - Tensor or LazyTensor wrapping matrix to add diagonal \
                                component to.
         - diag (scalar) - Scalar s so that s*I is added to the input matrix.
 
     Returns:
-        - matrix nxn - Variable or LazyTensor wrapping a new matrix with the diagonal \
+        - matrix nxn - Tensor or LazyTensor wrapping a new matrix with the diagonal \
                        component added.
     """
-    if not isinstance(diag, Variable):
-        raise RuntimeError("Expected a variable for the diagonal component.")
+    if not isinstance(diag, Tensor):
```

Comment:
Should we be using `torch.is_tensor` here instead? I remember there being some issues (back with PyTorch 0.2) with using `isinstance` with Tensors, but not sure if those have been fixed or not.

Reply:
yeah let's use `torch.is_tensor`